TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,166 @@
+    private TouchSensor limitUp;
+    private TouchSensor limitDown;
+    private
+    Map<String ,Double> liftPidPositions = new HashMap<>();
+
+    private SmartGamepad operator;
+    private org.firstinspires.ftc.teamcode.core.lib.pid.PIDController PIDController;
+
+    private Lift() {
+    }
+
+    /**
+     * Initialize method from the Subsystem Interface
+     * @param hardwareMap
+     * @param telemetry
+     */
+    @Override
+    public void initialize(HardwareMap hardwareMap, Telemetry telemetry) {
+        liftMotorRight = hardwareMap.get(DcMotor.class, LIFT_MOTOR_RIGHT);
+        liftMotorLeft = hardwareMap.get(DcMotor.class, LIFT_MOTOR_LEFT);
+        liftMotorLeft.setDirection(DcMotor.Direction.REVERSE);
+        liftMotorRight.setDirection(DcMotor.Direction.FORWARD);
+        limitUp = hardwareMap.get(TouchSensor.class, LIFT_LIMIT_UP);
+        limitDown = hardwareMap.get(TouchSensor.class, LIFT_LIMIT_DOWN);
+        this.telemetry = telemetry;
+
+        liftMotorLeft.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        liftMotorRight.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+        liftMotorLeft.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+        liftMotorRight.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
+
+        PIDController = new PIDController(PID.kP, PID.kI, PID.kD, PID.kF, POSITION);
+        PIDController.setTolerance(100);
+
+        liftPidPositions.put("Baixo",5712.0);
+        liftPidPositions.put("Medio",7140.0);
+        liftPidPositions.put("Alto", 8568.0);
+
+        telemetry.addData("LIFTSlide Subsystem", "Initialized");
+    }
+
+    /**
+     * Execute method from the Subsystem Interface
+     * @param gamepadManager
+     */
+    @Override
+    public void execute(GamepadManager gamepadManager) {
+        operator = gamepadManager.getOperator();
+
+        telemetry.addData("LIFTSlide Subsystem", "Running");
+
+        controlLIFTSlide();
+
+        if(isLimitDown()){liftPidPositions(0);}
+        
+        if(isLimitUP()){liftPidPositions(0);}
+
+        //Mantem posição após o operador soltar o botão
+        liftPidPositions(PIDController.calculate(liftMotorLeft.getCurrentPosition(),liftMotorLeft.getCurrentPosition()));
+
+        operator.whileButtonDPadUp()
+                .run(() -> {
+              liftPidPositions(PIDController.calculate(liftPidPositions.get("Baixo"), liftMotorLeft.getCurrentPosition()));
+                });
+
+        operator.whileButtonDPadDown()
+                .run(() -> {
+                    liftPidPositions(PIDController.calculate(liftPidPositions.get("Medio"),liftMotorLeft.getCurrentPosition()));
+                });
+
+        operator.whileButtonDPadRight()
+                .run(() -> {
+                    liftPidPositions(PIDController.calculate(liftPidPositions.get("Alto"),liftMotorLeft.getCurrentPosition()));
+                });
+
+        
+
+        
+        telemetry.addData("MR power", liftMotorRight.getPower());
```

Comment:
sugestion to all:
add a default start to when adding telemetry data
"[LIFT] + message"